### ISSUE TITLE
4 Tests in local_spec.rb are Failing

### ISSUE THREAD

**Issue by original-brownbear** [elastic/logstash#9566]
Same on Jenkins and local, these 4 fail 100% of the time currently:

```sh
rspec ./logstash-core/spec/logstash/config/source/local_spec.rb:309 # LogStash::Config::Source::Local when only the `config.string` is set returns a config
rspec ./logstash-core/spec/logstash/config/source/local_spec.rb:394 # LogStash::Config::Source::Local incomplete configuration when the output block is missing add stdout output
rspec ./logstash-core/spec/logstash/config/source/local_spec.rb:386 # LogStash::Config::Source::Local incomplete configuration when the input block is missing add stdin input
rspec ./logstash-core/spec/logstash/config/source/local_spec.rb:402 # LogStash::Config::Source::Local incomplete configuration when both the output block and input block are missing add stdin and output
```

Failing with:

```sh
Failures:

  1) LogStash::Config::Source::Local when only the `config.string` is set returns a config
     Failure/Error: expect(subject.pipeline_configs.first.config_string).to include(filter_block)
     
     ArgumentError:
       There should be exactly 1 unique protocol in config_parts. Found 2.
     # ./logstash-core/lib/logstash/config/pipeline_config.rb:17:in `initialize'
     # ./logstash-core/lib/logstash/config/source/local.rb:200:in `local_pipeline_configs'
     # ./logstash-core/lib/logstash/config/source/local.rb:162:in `pipeline_configs'
     # ./logstash-core/spec/logstash/config/source/local_spec.rb:310:in `block in (root)'
     # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
     # ./lib/bootstrap/rspec.rb:13:in `<main>'

  2) LogStash::Config::Source::Local incomplete configuration when the output block is missing add stdout output
     Failure/Error: expect(subject.pipeline_configs.first.config_string).to include(LogStash::Config::Defaults.output)
     
     ArgumentError:
       There should be exactly 1 unique protocol in config_parts. Found 2.
     # ./logstash-core/lib/logstash/config/pipeline_config.rb:17:in `initialize'
     # ./logstash-core/lib/logstash/config/source/local.rb:200:in `local_pipeline_configs'
     # ./logstash-core/lib/logstash/config/source/local.rb:162:in `pipeline_configs'
     # ./logstash-core/spec/logstash/config/source/local_spec.rb:395:in `block in (root)'
     # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
     # ./lib/bootstrap/rspec.rb:13:in `<main>'

  3) LogStash::Config::Source::Local incomplete configuration when the input block is missing add stdin input
     Failure/Error: expect(subject.pipeline_configs.first.config_string).to include(LogStash::Config::Defaults.input)
     
     ArgumentError:
       There should be exactly 1 unique protocol in config_parts. Found 2.
     # ./logstash-core/lib/logstash/config/pipeline_config.rb:17:in `initialize'
     # ./logstash-core/lib/logstash/config/source/local.rb:200:in `local_pipeline_configs'
     # ./logstash-core/lib/logstash/config/source/local.rb:162:in `pipeline_configs'
     # ./logstash-core/spec/logstash/config/source/local_spec.rb:387:in `block in (root)'
     # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
     # ./lib/bootstrap/rspec.rb:13:in `<main>'

  4) LogStash::Config::Source::Local incomplete configuration when both the output block and input block are missing add stdin and output
     Failure/Error: expect(subject.pipeline_configs.first.config_string).to include(LogStash::Config::Defaults.output, LogStash::Config::Defaults.input)
     
     ArgumentError:
       There should be exactly 1 unique protocol in config_parts. Found 2.
     # ./logstash-core/lib/logstash/config/pipeline_config.rb:17:in `initialize'
     # ./logstash-core/lib/logstash/config/source/local.rb:200:in `local_pipeline_configs'
     # ./logstash-core/lib/logstash/config/source/local.rb:162:in `pipeline_configs'
     # ./logstash-core/spec/logstash/config/source/local_spec.rb:403:in `block in (root)'
     # ./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
     # ./lib/bootstrap/rspec.rb:13:in `<main>'

Finished in 2.17 seconds (files took 7.33 seconds to load)
40 examples, 4 failures

```

**Comment by original-brownbear** [comment]
Seems this is caused by https://github.com/elastic/logstash/pull/9516

**Comment by original-brownbear** [comment]
@ycombinator can you take a look here, seems like we have to revert #9516 in `master` too :( It's breaking tests there like it did in `6.x` and it just went unnoticed due to #9567 :(

**Comment by ycombinator** [comment]
Ugh, sorry about this. Taking a look and reverting or fixing now.

**Comment by original-brownbear** [comment]
@ycombinator np + thanks :)

**Comment by ycombinator** [comment]
Resolved via #9568.
